lib/adapters/http.js

```diff
@@ -171,6 +171,13 @@ module.exports = function httpAdapter(config) {
       if (config.maxRedirects) {
         options.maxRedirects = config.maxRedirects;
       }
+      // Allow follow-redirects to resolve mixed protocol redirects if possible
+      if (config.httpAgent && config.httpsAgent) {
+        options.agents = {
+          http: config.httpAgent,
+          https: config.httpsAgent
+        };
+      }
```

Comment:
is it necessary for this setting to be here? since the last person added it above on lines 89-90